cmd/provisioner-localpv/app/provisioner.go

```diff
@@ -131,17 +140,34 @@ func (p *Provisioner) Delete(pv *v1.PersistentVolume) (err error) {
 	defer func() {
 		err = errors.Wrapf(err, "failed to delete volume %v", pv.Name)
 	}()
-
 	//Initiate clean up only when reclaim policy is not retain.
 	if pv.Spec.PersistentVolumeReclaimPolicy != v1.PersistentVolumeReclaimRetain {
 		//TODO: Determine the type of PV
 		pvType := GetLocalPVType(pv)
+		size := resource.Quantity{}
+		reqMap := pv.Spec.Capacity
+		if reqMap != nil {
+			size = pv.Spec.Capacity["storage"]
+		}
+
+		sendEventOrIgnore(pv.Name, size.String(), pvType, analytics.VolumeDeprovision)
 		if pvType == "local-device" {
 			return p.DeleteBlockDevice(pv)
 		}
 		return p.DeleteHostPath(pv)
 	}
-
 	glog.Infof("Retained volume %v", pv.Name)
 	return nil
 }
+
+func sendEventOrIgnore(pvName, capacity, stgType, method string) {
+	if menv.Truthy(menv.OpenEBSEnableAnalytics) {
+		analytics.New().Build().ApplicationBuilder().
+			SetVolumeType(stgType, method).
+			SetDocumentTitle(pvName).
+			SetLabel(analytics.EventLabelCapacity).
+			SetReplicaCount("1", method).
```

Comment:
Since method is getting passed into several methods, it will be good if you write a wrapper WithMethod(method) and access it accordingly.
Why ReplicaCount = 1 ?